test/UnitTest/Microsoft.AspNet.OData.Test.Shared/Query/Expressions/SelectExpandBinderTest.cs

```diff
@@ -1881,6 +1881,66 @@ public void CorrelatedSubqueryIncludesToListIfBufferingOptimizationIsTrueAndPage
+            var context = new ODataQueryContext(model, typeof(QueryPolygon)) { RequestContainer = new MockContainer() };
+            var binder = new SelectExpandBinder(settings, context);
+
+            var queryPolygonEntityType = model.SchemaElements.OfType<IEdmEntityType>().First(c => c.Name == "QueryPolygon");
+            var querySubCategoryEntityType = model.SchemaElements.OfType<IEdmEntityType>().First(c => c.Name == "QuerySubCategory");
+
+            // Arrange
+            Expression source = Expression.Constant(new QueryPolygon());
+
+            var propertyName = "Category";
+            IEdmNavigationProperty navProperty = queryPolygonEntityType.NavigationProperties().Single(c => c.Name == propertyName);
+            Assert.NotNull(navProperty);
+
+            // Act
+            Expression propertyValue = binder.CreatePropertyValueExpression(querySubCategoryEntityType, navProperty, source, null);
+
+            // Assert
+            Assert.Equal(ExpressionType.MemberAccess, propertyValue.NodeType);
+            Assert.Equal(TypeHelper.GetProperty(typeof(QueryPolygon), propertyName), (propertyValue as MemberExpression).Member);
+            Assert.Equal(String.Format("({0} As QueryPolygon).{1}", source.ToString(), propertyName), propertyValue.ToString());
+        }
+
+        [Fact]
+        public void CreatePropertyValueExpression_DerivedNavigationPropertyRedeclaredInSubType_ReturnsMemberAccessExpression()
+        {
+            var modelBuilder = new ODataConventionModelBuilder();
+            modelBuilder.EntitySet<QueryPentagon>("Shapes");
+            var model = modelBuilder.GetEdmModel();
+
+            var settings = new ODataQuerySettings { HandleNullPropagation = HandleNullPropagationOption.False };
+            var context = new ODataQueryContext(model, typeof(QueryPentagon)) { RequestContainer = new MockContainer() };
+            var binder = new SelectExpandBinder(settings, context);
+
+            var queryPentagonEntityType = model.SchemaElements.OfType<IEdmEntityType>().First(c => c.Name == "QueryPentagon");
+            var querySubCategoryEntityType = model.SchemaElements.OfType<IEdmEntityType>().First(c => c.Name == "QuerySubCategory");
+
+            // Arrange
+            Expression source = Expression.Constant(new QueryPentagon());
+
+            var propertyName = "Category";
+            IEdmNavigationProperty navProperty = queryPentagonEntityType.NavigationProperties().Single(c => c.Name == propertyName);
+            Assert.NotNull(navProperty);
+
+            // Act
+            Expression propertyValue = binder.CreatePropertyValueExpression(querySubCategoryEntityType, navProperty, source, null);
+
+            // Assert
+            Assert.Equal(ExpressionType.MemberAccess, propertyValue.NodeType);
+            Assert.Equal(TypeHelper.GetProperty(typeof(QueryPentagon), propertyName), (propertyValue as MemberExpression).Member);
```

Comment:
Here you're verifying that the property type is `QuerySubCategory` and not `QueryCategory` since that's the type of `QueryPolygon.Category`?

Reply:
Correct